pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -131,12 +132,26 @@ func (gce *GCECloud) DetachDisk(devicePath string, nodeName types.NodeName) erro
 	}
 
 	mc := newDiskMetricContext("detach", inst.Zone)
-	detachOp, err := gce.service.Instances.DetachDisk(gce.projectID, inst.Zone, inst.Name, devicePath).Do()
+	detachOp, err := gce.getDetachDiskOp(inst.Zone, inst.Name, devicePath)
 	if err != nil {
 		return mc.Observe(err)
 	}
 
-	return gce.waitForZoneOp(detachOp, inst.Zone, mc)
+	return gce.manager.WaitForZoneOp(detachOp, inst.Zone, mc)
+}
+
+// Returns a version agnostic DetachDisk operation.
+func (gce *GCECloud) getDetachDiskOp(
```

Comment:
Can you add DetachDisk to the ServiceManager, so that it also follows the same pattern as the rest?